### PR TITLE
Completed ex5

### PR DESCRIPTION
**Pull request by Alishow1111**

### CODE REVIEW

File: src/components/ArticleCard.jsx [hunk 1]
```diff
@@ -1,9 +1,15 @@
 import Button from 'react-bootstrap/Button';
 import Card from 'react-bootstrap/Card';
 import Col from 'react-bootstrap/Col';
+import {Link } from "react-router-dom";
+
 
 
 function ArticleCard({article}) {
+
+  function handleClick(){
```
Comment (2202Hannah) [inline src/components/ArticleCard.jsx:10]: Please remove this empty handleClick if it's not needed

File: src/components/ArticleCard.jsx [hunk 2]
```diff
@@ -16,7 +22,9 @@ function ArticleCard({article}) {
             <Card.Text>
             Topic: {article.topic}
             </Card.Text>
-            <Button variant="primary">Read More...</Button>
+            <Link to={`/articles/${article.article_id}`}>
```
Comment (2202Hannah) [inline src/components/ArticleCard.jsx:25]: Nice use of Link to direct the user to the appropriate route for the article they have selected

File: src/components/ArticlePage.jsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from 'react'
+import { useParams } from 'react-router-dom';
+import {fetchArticleById} from './utils/api';
+import Spinner from 'react-bootstrap/Spinner';
+
+
+
+function ArticlePage() {
+    const {article_id} = useParams();
```
Comment (2202Hannah) [inline src/components/ArticlePage.jsx:9]: Nice use of useParams() 👍

File: src/components/ArticlePage.jsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    useEffect(() => {
+        fetchArticleById(article_id)
+        .then((articleResponse) => {
+            setArticle(articleResponse);
+            setLoading(false)
+        })
+    }, [])
+
+    if (loading){
```
Comment (2202Hannah) [inline src/components/ArticlePage.jsx:22]: Great user feedback with this loading state and conditional rendering :)

**Review by 2202Hannah — APPROVED** [review]
Great work on task 5! Your app is looking good and the user experience is good, I particularly like the loading spinners. The user is able to navigate to the individual article both by clicking on the article card and via the URL. You can merge to main when you are ready.

File: src/components/ArticleCard.jsx (hunk 2 above)
Comment (Alishow1111) [reply src/components/ArticleCard.jsx:25]: Thanks

**Pull request merged**